File: waste-reco-api/app/models/reporte.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, Numeric
from models.base import Base

class Reporte(Base):
    __tablename__ = "residuos"

    id = Column(Integer, primary_key=True, index=True)
    precision_usuario = Column(Numeric(5, 4), index=True)
    tasa_acierto = Column(Numeric(5, 4), index=True)
    fecha_reporte = Column(String, index=True)
    id_residuo = Column(Integer, ForeignKey("residuos.id"), index=True)
    id_usuario = Column(Integer, ForeignKey("usuario.id"), index=True)

File: waste-reco-web/src/models/Reporte.ts
import type { Residuo } from './Residuo'
import type Usuario from './Usuario'

export default interface Reporte {
  id: number
  precision_usuario: number
  tasa_acierto: number
  fecha_reporte: string
  id_residuo: number
  id_usuario: number
}

export interface ReporteResponse {
  id: number
  precision_usuario: number
  tasa_acierto: number
  fecha_reporte: string
  residuo: Residuo
  usuario: Usuario
}
